### ISSUE TITLE
Cannot play from 'Remote' app on iOS

### ISSUE THREAD

**Issue by buschschwick** [ejurgensen/forked-daapd#173]
I'm currently experiencing an issue where forked-daapd works on all my OSX machines, but when I try to play music from the Remote app on iOS, I can see my library but when selecting a track it does not play. I checked the logs and the only thing I have are these two lines:

`Player: cannot start playback: no output started`

&

`dacp: could not start playback`

Any thoughts?

**Comment by buschschwick** [comment]
And as a note: it has worked but has recently stopped.

**Comment by ejurgensen** [comment]
If it is local audio the good old reboot might help. If it doesn't, or if you also have AirPlay devices, then please submit the log lines with debug log level.

**Comment by buschschwick** [comment]
I'm kind of new to this, is the debug log level different then the standard log? The first two errors are the only thing I see in the log file. Reboot did not fix the issue, however I did notice that my AppleTV turns on when i try to play from my iPhone Remote app, but the music does not play in either place. 

Is it possible that the AirPlay function is somehow hijacking it, or something along those lines?  I've never set up my AppleTV to play nice with forked-daapd.

**Comment by ejurgensen** [comment]
Yes, you set the log level to debug in the config file. It is very verbose, so set it back afterwards, and please don't post the entire log here, just the section from right after you start playback.

You don't need to set up the AppleTV for forked-daapd. AirPlay should work by itself. It the Apple TV the only AirPlay device you have, or do you have others? (so you can see if they work)

**Comment by buschschwick** [comment]
Hello again, yes my apple tv is my only airplay device. Here is my log file, its rather large even though i'm only starting from the playback of the file itself...
```
[2015-06-29 19:46:35] [ INFO]   player: Opening 'Daylight' (/home/ericstout/Music/Aesop Rock/Daylight/01 Daylight.mp3)
[2015-06-29 19:46:35] [DEBUG]   ffmpeg: Probed with size=8192 and score=51
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker=db avail_size_in_buf=5391
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: index=0
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: qscale[0]: 3
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker parser used 67 bytes (536 bits)
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker=db avail_size_in_buf=5322
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: index=1
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: qscale[1]: 6
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker parser used 67 bytes (536 bits)
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker=c0 avail_size_in_buf=5253
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: sof0: picture: 200x198
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: component 0 2:2 id: 0 quant:0
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: component 1 1:1 id: 1 quant:1
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: component 2 1:1 id: 2 quant:1
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: pix fmt id 22111100
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker parser used 17 bytes (136 bits)
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker=c4 avail_size_in_buf=5234
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: class=0 index=0 nb_codes=12
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker parser used 31 bytes (248 bits)
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker=c4 avail_size_in_buf=5201
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: class=1 index=0 nb_codes=251
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker parser used 181 bytes (1448 bits)
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker=c4 avail_size_in_buf=5018
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: class=0 index=1 nb_codes=12
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker parser used 31 bytes (248 bits)
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker=c4 avail_size_in_buf=4985
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: class=1 index=1 nb_codes=251
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker parser used 181 bytes (1448 bits)
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: escaping removed 87 bytes
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker=da avail_size_in_buf=4802
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: component: 0
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: component: 1
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: component: 2
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker parser used 4714 bytes (37707 bits)
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: marker=d9 avail_size_in_buf=0
[2015-06-29 19:46:36] [DEBUG]   ffmpeg: mjpeg decode frame unused 0 bytes
[2015-06-29 19:46:36] [ WARN]   ffmpeg: max_analyze_duration reached
[2015-06-29 19:46:36] [ WARN]   ffmpeg: Estimating duration from bitrate, this may be inaccurate
[2015-06-29 19:46:36] [DEBUG]       db: Running query 'SELECT COUNT(*) FROM files f WHERE f.disabled = 0 AND id = 1756;'
[2015-06-29 19:46:36] [DEBUG]       db: Starting query 'SELECT f.* FROM files f WHERE f.disabled = 0 AND id = 1756 ;'
[2015-06-29 19:46:36] [DEBUG]       db: Running query 'SELECT f.* FROM files f WHERE f.id = 1756;'
[2015-06-29 19:46:36] [DEBUG]  artwork: Artwork request for item 1756 (01 Daylight.mp3)
[2015-06-29 19:46:36] [DEBUG]  artwork: Looking for artwork for group with persistentid 5029747732759618709
[2015-06-29 19:46:36] [DEBUG]    cache: Running query 'SELECT a.format, a.data FROM artwork a WHERE a.type = 0 AND a.persistentid = 5029747732759618709 AND a.max_w = 600 AND a.max_h = 600;'
[2015-06-29 19:46:36] [DEBUG]    cache: Cache hit: SELECT a.format, a.data FROM artwork a WHERE a.type = 0 AND a.persistentid = 5029747732759618709 AND a.max_w = 600 AND a.max_h = 600;
[2015-06-29 19:46:36] [DEBUG]  artwork: Group 5029747732759618709 found in cache with format 2
[2015-06-29 19:46:36] [DEBUG]     raop: Building OPTIONS for Apple TV
[2015-06-29 19:46:36] [ INFO]   player: Autoselecting AirPlay device Apple TV
[2015-06-29 19:46:36] [ WARN]    event: event_del: event has no event_base set.
[2015-06-29 19:46:36] [DEBUG]     raop: Local address: 10.0.1.9 (LL: no) port 50522
[2015-06-29 19:46:36] [DEBUG]     raop: Building ANNOUNCE for Apple TV
[2015-06-29 19:46:36] [DEBUG]     raop: Building SETUP for Apple TV
[2015-06-29 19:47:08] [  LOG]     raop: SETUP request failed in session startup: 500 Internal Server Error
[2015-06-29 19:47:08] [  LOG]   player: Cannot start playback: no output started
[2015-06-29 19:47:08] [  LOG]     dacp: Could not start playback
[2015-06-29 19:47:08] [DEBUG]   player: Player status: stopped
[2015-06-29 19:47:08] [DEBUG]    httpd: Found query string
[2015-06-29 19:47:08] [DEBUG]     dacp: DACP request: /ctrl-int/1/playstatusupdate?revision-number=3&session-id=112462279
[2015-06-29 19:47:08] [DEBUG]    httpd: Found query string
[2015-06-29 19:47:08] [DEBUG]     dacp: DACP request: /ctrl-int/1/playqueue-contents?span=50&session-id=112462279
[2015-06-29 19:47:08] [DEBUG]     dacp: Fetching playqueue contents
[2015-06-29 19:47:08] [DEBUG]   player: Player status: stopped
[2015-06-29 19:47:08] [DEBUG]    httpd: Found query string
[2015-06-29 19:47:08] [DEBUG]     dacp: DACP request: /ctrl-int/1/getproperty?properties=dmcp.volume&session-id=112462279
[2015-06-29 19:47:08] [DEBUG]   player: Player status: stopped
[2015-06-29 19:47:08] [DEBUG]    httpd: Found query string
[2015-06-29 19:47:08] [DEBUG]     dacp: DACP request: /ctrl-int/1/getspeakers?session-id=112462279
[2015-06-29 19:47:09] [DEBUG]    httpd: Found query string
[2015-06-29 19:47:09] [DEBUG]     dacp: DACP request: /ctrl-int/1/getproperty?properties=dacp.playerstate,dacp.nowplaying,dacp.playingtime,dmcp.volume,dacp.volumecontrollable,dacp.availableshufflestates,dacp.availablerepeatstates,dacp.shufflestate,dacp.repeatstate,dacp.fullscreenenabled,dacp.fullscreen,dacp.visualizerenabled,dacp.visualizer,com.apple.itunes.itms-songid,com.apple.itunes.has-chapter-data,com.apple.itunes.mediakind,com.apple.itunes.extended-media-kind&session-id=112462279
[2015-06-29 19:47:09] [DEBUG]   player: Player status: stopped
```

It looks like its auto selecting to play with the apple tv and waking it from sleep to play, but does nothing. 

Thanks for your help!

**Comment by ejurgensen** [comment]
The noteworthy part is this:
[2015-06-29 19:46:36] [DEBUG]     raop: Building SETUP for Apple TV
[2015-06-29 19:47:08] [  LOG]     raop: SETUP request failed in session startup: 500 Internal Server Error

This is forked-daapd making a setup request to the ATV and the ATV returning (after 30 seconds?) with an internal error. I don't know what the cause of that would be. Have you upgraded your ATV recently?

**Comment by buschschwick** [comment]
Yes, my Apple TV is up to date. There aren't any passwords on my Airplay denying service or anything like that. Is there a way to just disable airplay, since I don't really use it anyway? 

I'm not sure where to go from here.

**Comment by ejurgensen** [comment]
What I meant was if the problem started after an upgrade. I don't understand what disabling AirPlay would achieve?

**Comment by buschschwick** [comment]
Ah, sorry misunderstood the update part. I'm not sure if it updated or not to be honest, sorry. The reason I bring up disabling AirPlay is because I'm never telling it to airplay, the AppleTV even when unplugged from power is being pushed as the default player - see this section of the log:

```
[2015-07-01 17:38:49] [DEBUG]     raop: Building OPTIONS for Apple TV
[2015-07-01 17:38:49] [ INFO]   player: Autoselecting AirPlay device Apple TV
[2015-07-01 17:39:07] [  LOG]     raop: OPTIONS request failed in session startup: 0 (null)
[2015-07-01 17:39:07] [  LOG]   player: Cannot start playback: no output started
[2015-07-01 17:39:07] [  LOG]     dacp: Could not start playback
[2015-07-01 17:39:07] [DEBUG]   player: Player status: stopped
```
I'm not sure why its auto selecting it, especially when it's not even available.

**Comment by buschschwick** [comment]
So another thing to note, is even with the AppleTV unplugged it was still trying to push the audio there. So I restarted my Ubuntu machine and re-launched forked-daapd while the Apple TV was still unplugged. Success, the audio now plays TO my Ubuntu machine when selecting songs from the Remote IOS app. I can even plug the AppleTV back in and it continues to work fine.

So then I tried relaunching forked-daapd again just to see if the issue would re-appear, and yes it does. I'd like to try to avoid having to unplug my AppleTV everytime I restart my machine. 

Just giving out all the info I know, any thoughts? Thanks a ton for your help by the way.
